src/main/java/dev/morling/onebrc/CalculateAverage_lawrey_magic.java

```diff
@@ -0,0 +1,225 @@
+/*
+ *  Copyright 2023 The original authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package dev.morling.onebrc;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteOrder;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.LongStream;
+
+public class CalculateAverage_lawrey_magic {
+
+    // Path to the file containing temperature measurements.
+    private static final String FILE = "./measurements.txt";
+    private static final String[] stations = new String[6122];
```

Comment:
This fails the tests! There can be 10,000 different stations. See `./test.sh lawrey_magic`.

Reply:
Note we have added another test since then which covers longer station names. I.e. please rebase to latest main to be sure to run all the tests. Thx!